h5py/_npystrings.pyx

```diff
@@ -85,13 +86,16 @@ cdef herr_t npystrings_pack_cb(
 ) except -1:
     cdef npystrings_pack_t* info = <npystrings_pack_t*>operator_data
     cdef const char* buf = info[0].contig[info[0].i]
-    # Deep copy char* from h5py into the NumPy array
-    res = NpyString_pack(
-        info[0].allocator,
-        <npy_packed_static_string*>elem,
-        buf,
-        strlen(buf),
-    )
+    if buf is NULL:
+        NpyString_pack_null(info[0].allocator, <npy_packed_static_string*>elem)
```

Comment:
```suggestion
        # Special case of default values created by resize(). Regular empty strings
        # are valid char* with strlen(buf) == 0.
        NpyString_pack_null(info[0].allocator, <npy_packed_static_string*>elem)
```

Reply:
Aha, thank you, I totally overlooked that the return value was doing something. :sweat_smile: